### ISSUE TITLE
rowDelimiter 'auto' option not working as expected

### ISSUE THREAD

**Issue by cookch10** [wdavidw/node-csv-parse#41]
I just started using node-csv, and am using the parser with the following options:

```javascript
var csvParser = csv.parse({
    auto_parse: true, // ensures that numeric values remain numeric
    columns: null,
    delimiter: ',',
    quote: '',
    relax: true,
    rowDelimiter: 'auto', // this is an issue, I had to set \n here as 'auto' wasn't working, nor was 'windows'.
    skip_empty_lines: false
});
```
According to the [documentation](http://csv.adaltas.com/parse/):  `rowDelimiter (chars|constant)`:
String used to delimit record rows or a special value; special constants are 'auto', 'unix', 'mac', 'windows', 'unicode'; defaults to 'auto' (discovered in source or 'unix' if no source is specified).

I tried using 'auto' on a data file that uses `\n` line endings, but it wasn't parsing correctly.  By setting the rowDelimiter parse option to 'auto', is it supposed to perform some auto-detection of line endings?

**Comment by wdavidw** [comment]
you'll need to provide me with either a sample or test that illustrate your issue. This functionnality is covered by tests and I wont be able to reproduce your issue otherwise.

**Comment by cookch10** [comment]
Here is a [link](https://dl.dropboxusercontent.com/u/8720380/tmp/sample.csv) to a sample delimited text file that does not parse properly with the `rowDelimiter` option set to `auto`.  I'm running Windows 7x64 OS with Node v.0.12.2.

Here's a sample code snippet of how I'm pulling in the file:

```javascript
var csv = require('csv');

var inputCSVFilePath = 'sample.csv';

var inputCSVFileStream = _fs.createReadStream(inputCSVFilePath);

var outputArr = [];

var csvParseOptions = {
    auto_parse: true, // ensures that numeric values remain numeric
    columns: null, // setting this to true will create an array of objects, otherwise it will be an array of arrays rows[(columns[])].
    delimiter: ',',
    quote: '',
    relax: true,
    rowDelimiter: 'auto', // this is an issue, I had to set the \n here as 'auto' wasn't working, nor was 'windows'.  Maybe look at auto-detecting line endings?
    skip_empty_lines: false
};

var csvParser = csv.parse(csvParseOptions);

csvParser.on('error', function (error) {
    throw error;
});

csvParser.on('readable', function () {
    var record;

    while (record = csvParser.read()) {
        if (!csvParser.lines) {
            // placeholder:  This is the column header, add additional column headers here
        } else {
            // placeholder:  This is a row of data, add the appropriate field data here aligning with new column headers
        }

        outputArr.push(record);
    }
});

csvParser.on('finish', function () {
});

csvParser.on('end', function () {
    console.log(outputArr);
});

inputCSVFileStream.pipe(csvParser);
```

**Comment by wdavidw** [comment]
did u try without "relax: true" which is buggy?

**Comment by cookch10** [comment]
I did.  The result is the same with `relax: true`, `relax: false`, and `relax` not set at all.

**Comment by ebbersjathomes** [comment]
I'm having the same issue. When I specify a rowDelimiter from the constant value list it doesn't parse new lines correctly Example:

```javascript
'use strict';

var generate = require('csv-generate'),
    parse = require('csv-parse'),
    transform = require('stream-transform');

var generator = generate({columns: 5, length: 5, delimiter: '|'}),
    parser = parse({delimiter: '|', rowDelimiter: 'auto'}),
    transformer = transform(function(record, callback){
        console.log(record.length);
        callback(null, null);
    });

generator.pipe(parser).pipe(transformer);
```
21
```javascript
'use strict';

var generate = require('csv-generate'),
    parse = require('csv-parse'),
    transform = require('stream-transform');

var generator = generate({columns: 5, length: 5, delimiter: '|'}),
    parser = parse({delimiter: '|'}),
    transformer = transform(function(record, callback){
        console.log(record.length);
        callback(null, null);
    });

generator.pipe(parser).pipe(transformer);
```
5
5
5
5
5

**Comment by wdavidw** [comment]
i just published a new 0.1.0 version. My mistake, i sometime forget that pushing to npm isnt the same as pushing to git.
